src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetBuildManifestMetadataTests.cs

```diff
@@ -0,0 +1,115 @@
+            data.Item2.Should().BeEquivalentTo(SharedObjects.ExpectedSigningInfo.Concat(SharedObjects.ExpectedSigningInfo2));
+            SharedObjects.CompareManifestBuildData(data.Item3, SharedObjects.ExpectedManifestBuildData);
+        }
+
+        [Test]
+        public void GivenFileThatIsNotAManifest_ExceptionExpected()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "NonManifest", new CancellationToken());
+            act.Should().Throw<InvalidOperationException>().WithMessage("There is an error in XML document (1, 1).");
+        }
+
+        [Test]
+        public void GivenBadlyFormattedXml_ExceptionExpected()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "BadXml", new CancellationToken());
+            act.Should().Throw<InvalidOperationException>().WithMessage("There is an error in XML document (2, 1).");
+        }
+
+        [Test]
+        public void GivenAnEmptyManifest_ExceptionExpected()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "EmptyManifest", new CancellationToken());
+            act.Should().Throw<NullReferenceException>();
+        }
+
+        [Test]
+        public void GivenManifestWithoutPackages()
+        {
+            var data = pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "ManifestWithoutPackages", new CancellationToken());
+            data.Item1.Should().BeEquivalentTo(SharedObjects.ExpectedNoPackagesManifestMetadata);
+            data.Item2.Should().BeEquivalentTo(SharedObjects.ExpectedSigningInfo);
+            SharedObjects.CompareManifestBuildData(data.Item3, SharedObjects.ExpectedManifestBuildData);
+        }
+
+        [Test]
+        public void GivenManifestWithUnversionedPackage()
+        {
+            var data = pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "UnversionedPackage", new CancellationToken());
+            data.Item1.Should().BeEquivalentTo(SharedObjects.ExpectedUnversionedPackagedManifestMetadata);
+        }
+
+        [Test]
+        public void GivenManifestWithoutBlobs()
+        {
+            var data = pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "ManifestWithoutBlobs", new CancellationToken());
+            data.Item1.Should().BeEquivalentTo(SharedObjects.ExpectedNoBlobManifestMetadata);
+            data.Item2.Should().BeEquivalentTo(SharedObjects.ExpectedSigningInfo);
+            SharedObjects.CompareManifestBuildData(data.Item3, SharedObjects.ExpectedManifestBuildData);
+        }
+
+        // TODO: This gets a versionId from another method, which needs to be overwritten with DI. Turning it off while I get DI set up for the project.
+        [Test]
+        [Ignore("Requires DI that isn't set up for the project yet.")]
```

Comment:
Let's get this working before we do a merge.

Reply:
That's the plan, just getting a jump on the review timer. :)